e2e/cli-e2e/tests/collect.spec.ts

```diff
@@ -23,58 +28,70 @@ describe('CLI collect', () => {
     });
 
   const cliPath = join('..', '..', 'dist', 'packages', 'cli');
+  const reportPath = join('tmp', 'react-todos-app');
+  const reportFile = (filename: string, ext = 'json') =>
+    join(reportPath, `${filename}.${ext}`);
+
+  const filename = () =>
+    reportNameFromReport({ date: new Date().toISOString() });
+  const baseArgs = [cliPath, 'collect', '--verbose', '--no-progress'];
 
   beforeEach(async () => {
     vi.clearAllMocks();
-    cleanFolderPutGitKeep();
-  });
-
-  afterEach(() => {
-    cleanFolderPutGitKeep();
+    setupFolder();
   });
 
   it('should run ESLint plugin and create report.json', async () => {
+    const reportFileName = filename();
     const { code, stderr } = await executeProcess({
       command: 'npx',
-      args: [cliPath, 'collect', '--no-progress'],
+      args: [
+        ...baseArgs,
+        '--persist.format=json',
```

Comment:
I would omit the explicit format, because I think it's valuable to test that the JSON file gets created regardless. E2E tests should focus on covering critical paths, i.e. what's most commonly used (users usually won't specify JSON format explicitly).

```suggestion
```